File: wordlist.rb
#!/usr/bin/env ruby

# make sorted word frequency list from text

require "unicode_utils"  # allows downcasing
require "unicode"        # allows sorting

diff_re = /^(?:\+{3}|-|@@|\p{Space}(?:\p{Space}\p{Space})*(?=\P{Space}))/
word_count = Hash.new(0)
word_re = /#?(?:\p{Alnum}|-)*\p{Letter}{2,}(?:\p{Alnum}|-)*/u
ARGF.each do |line|
  next if diff_re.match line
  line.strip!
  line.scan(word_re) do |w|
    next if (w =~ /^#/)
    word_count[UnicodeUtils.downcase(w)] += 1
  end
end

word_count.keys.sort{|w,v|
  Unicode.strcmp(w,v)
}.each do |w|
  puts "%4d %s" % [word_count[w], w]
end

File: wordlist.pl
#!/usr/bin/env perl 
#===============================================================================
#
#         FILE: wordlist.pl
#
#        USAGE: ./wordlist.pl  
#
#  DESCRIPTION: make word frequency list from text
#
#      CREATED: 24.03.2013 22:47:23
#  Last Change: 2013-03-27, 21:41:48 CET
#===============================================================================

# BOILERPLATE:
use strict;
use warnings;
use utf8;
# binmode(STDIN,":utf8");
# binmode(STDERR,":utf8");
# binmode(STDOUT,":utf8");
use open qw( :encoding(UTF-8) :std );
# use charnames qw( :full :short );
# use feature qw(say state switch unicode_strings);
use feature qw(unicode_strings);
use re "/u";
use autodie;
use v5.14;



# SORTING SET UP:
#
# without locale:
# use Unicode::Collate;
# my $Collator = Unicode::Collate->new(); # unbreak sorting on Mac
# alternatively, with Locale:
use Unicode::Collate::Locale;
my $Collator = Unicode::Collate::Locale->new(locale => "de_DE"); # unbreak sorting on Mac
my $sort = $Collator->sort;

# COUNT
my %words;
while (<>){
    next if (m/^(?:@@|\+{3}|\-|\s(?:\s\s)*(?=\S))/);
    while (m/#?(?:\w|-)*\p{L}{2,}(?:\w|-)*/g){
        next if ($& =~ /^#/); # skip IDs
        $words{lc($&)}++;
    }
}

# SORT & PRINT (1)
for my $word ($Collator->sort(keys(%words))){
    printf("%4d %s\n", $words{$word}, $word);
}

# SORT & PRINT (2)
# # for non-Macs with working locale:
# use locale;
# for my $word (sort(keys(%words))){
#     printf("%4d %s\n", $words{$word}, $word);
# }
